src/ring_theory/polynomial/hermite.lean

```diff
@@ -124,5 +137,72 @@ begin
 end
 
 end coeff
+end polynomial
+
+/-! ### Lemmas about `polynomial.hermite_gauss` -/
+
+section gaussian
+
+/-- The real Gaussian function -/
+def gaussian : ℝ → ℝ := λ x, real.exp (-(x^2 / 2))
+
+lemma inv_gaussian_eq : gaussian⁻¹ = λ x, real.exp (x^2 / 2) :=
+by { ext, simp [gaussian, real.exp_neg] }
+
+lemma inv_gaussian_mul_gaussian (x : ℝ) : gaussian⁻¹ x * gaussian x = 1 :=
+by rw [inv_gaussian_eq, gaussian, ← real.exp_add, add_neg_self, real.exp_zero]
+
+lemma deriv_gaussian (x : ℝ) : deriv gaussian x = -x * gaussian x :=
+by simp [gaussian, mul_comm]
+
+lemma deriv_inv_gaussian (x : ℝ) : deriv gaussian⁻¹ x = x * gaussian⁻¹ x :=
+by simp [inv_gaussian_eq, mul_comm]
+
+lemma cont_diff_gaussian : cont_diff ℝ ⊤ gaussian :=
+((cont_diff_id.pow 2).div_const 2).neg.exp
+
+lemma cont_diff.iterated_deriv :
+∀ (n : ℕ) (f : ℝ → ℝ) (hf : cont_diff ℝ ⊤ f), cont_diff ℝ ⊤ (deriv^[n] f)
+| 0     f hf := hf
+| (n+1) f hf := cont_diff.iterated_deriv n (deriv f) (cont_diff_top_iff_deriv.mp hf).2
+
+/-- The Gaussian form of `hermite n` -/
+def hermite_gauss (n : ℕ) : ℝ → ℝ :=
+λ x, (-1)^n * (gaussian⁻¹ x) * (deriv^[n] gaussian x)
+
+lemma hermite_gauss_def (n : ℕ) :
+hermite_gauss n = λ x, (-1)^n * (gaussian⁻¹ x) * (deriv^[n] gaussian x) := rfl
+
+lemma hermite_gauss_succ (n : ℕ) : hermite_gauss (n+1)
+= id * (hermite_gauss n) - deriv (hermite_gauss n):=
+begin
+  ext,
+  simp only [hermite_gauss, function.iterate_succ', function.comp_app,
+             id.def, pi.mul_apply, pi.sub_apply, pow_succ],
+  rw [deriv_mul, deriv_const_mul, deriv_inv_gaussian],
+  ring,
+  { simp [inv_gaussian_eq] },
+  { simp [inv_gaussian_eq] },
+  { apply (cont_diff_top_iff_deriv.mp (cont_diff.iterated_deriv _ _ cont_diff_gaussian)).1 }
+end
```

Comment:
@lukemantle and I looked at what this becomes if the auxiliary definitions `gaussian` and `hermite_gauss` are removed. Here's a statement that only uses the new `cont_diff.iterated_deriv` lemma above. It's less legible but works OK.

```
lemma hermite_exp_succ (n : ℕ) : 
(λ x, (-1 : ℝ)^(n+1) * real.exp (x^2 / 2) * (deriv^[n+1] (λ y, real.exp (-(y^2 / 2))) x))
= id * (λ x, (-1 : ℝ)^n * real.exp (x^2 / 2) * (deriv^[n] (λ y, real.exp (-(y^2 / 2))) x)) - 
  deriv (λ x, (-1 : ℝ)^n * real.exp (x^2 / 2) * (deriv^[n] (λ y, real.exp (-(y^2 / 2))) x)) :=
begin
  ext,
  simp only [function.iterate_succ', id.def, pi.mul_apply, 
             function.comp_app, pi.sub_apply],
  rw [deriv_mul, deriv_const_mul,
      (by simp [mul_comm] : deriv (λ y, real.exp (y^2 / 2)) x = x * real.exp (x^2 / 2))],
  ring_exp,
  { simp },
  { simp },
  { apply (cont_diff_top_iff_deriv.mp (cont_diff.iterated_deriv _ _ _)).1,
    exact ((cont_diff_id.pow 2).div_const 2).neg.exp, }
end
```

Reply:
This proof circumvents `gaussian, inv_gaussian_eq, inv_gaussian_mul_gaussian, deriv_gaussian, deriv_inv_gaussian, cont_diff_gaussian, hermite_gauss, hermite_gauss_def`.
(Note: `deriv_inv_gaussian` and `cont_diff_gaussian` are still "used" but written out manually.)

Reply:
So the question is whether it's better this way (with a less legible statement) or if we like statements like
```
  ∀ (n : ℕ) (x : ℝ), eval x (map (algebra_map ℤ ℝ) (hermite n)) =
    (-1)^n * (gaussian⁻¹ x) * (deriv^[n] gaussian x) := sorry
```

Reply:
(removed `gaussian` in the end)